source/PlayerInfo.cpp

```diff
@@ -910,6 +910,34 @@ void PlayerInfo::DoAccounting()
 		message += ".";
 		Messages::Add(message, Messages::Importance::High, true);
 		accounts.AddCredits(salariesIncome + tributeIncome + balance.assetsReturns);
+
+		if(tributeIncome)
+		{
+			// Apply reputation penalties for dominated planets.
+			set<const Government *> governments;
+			for(const auto &it : tributeReceived)
+			{
+				if(it.first->TributeDailyPenalty())
+				{
+					const Government *gov = it.first->GetGovernment();
+					gov->AddReputation(-it.first->TributeDailyPenalty());
+					governments.insert(gov);
+				}
+			}
+			message = "Your reputation with ";
+			if(governments.size() == 1)
+				message += (*governments.cbegin())->GetName();
+			else
+				for(auto it = governments.cbegin(); it != governments.cend(); ++it)
+				{
+					if(it == prev(governments.cend()))
+						message += "and " + (*it)->GetName();
+					else
+						message += (*it)->GetName() + ", ";
+				}
+			message += " has been severed because of your active tribute demands.";
```

Comment:
```suggestion
			message = "You have lost reputation with ";
			if(governments.size() == 1)
				message += (*governments.cbegin())->GetName();
			else
				for(auto it = governments.cbegin(); it != governments.cend(); ++it)
				{
					if(it == prev(governments.cend()))
						message += "and " + (*it)->GetName();
					else
						message += (*it)->GetName() + ", ";
				}
			message += " due to active tributes.";
```

Reply:
Perhaps there should be a 'the' after the first bit, you'll have to check which looks better.

Reply:
Koranir's change with the's - they do sound better.
```suggestion
			message = "You have lost reputation with the ";
			if(governments.size() == 1)
				message += (*governments.cbegin())->GetName();
			else
				for(auto it = governments.cbegin(); it != governments.cend(); ++it)
				{
					if(it == prev(governments.cend()))
						message += "and the " + (*it)->GetName();
					else
						message += (*it)->GetName() + ", ";
				}
			message += " due to active tributes.";
```